src/shared/generator.c

```diff
@@ -64,6 +64,7 @@ static int write_fsck_sysroot_service(const char *dir, const char *what) {
                 "Documentation=man:systemd-fsck-root.service(8)\n"
                 "Description=File System Check on %2$s\n"
                 "DefaultDependencies=no\n"
+                "ConditionPathExists=%2$s\n"
```

Comment:
That does not look right. It does not ensure that root path exists, but rather (silently) skips executing this service when path does not exist. This unit already has `After=initrd-root-device.target` so if you miss root device here, unit that creates it is not properly ordered before `initrd-root-device.target` and you need to fix it, not workaround here.